dask/array/core.py

```diff
@@ -1059,14 +1059,17 @@ def elemwise(op, *args, **kwargs):
     arrays = [arg for arg in args if isinstance(arg, Array)]
     other = [(i, arg) for i, arg in enumerate(args) if not isinstance(arg, Array)]
 
-    if not all(a._dtype is not None for a in arrays):
+    if 'dtype' in kwargs:
+        dt = kwargs['dtype']
+    elif not all(a._dtype is not None for a in arrays):
```

Comment:
This is the same as `elif any(a._dtype is None for a in arrays)`, I think?

That would avoid the double-negative